common/cuda_hip/preconditioner/batch_scalar_jacobi.hpp.inc

```diff
@@ -65,12 +104,30 @@ public:
         size_type,
         const gko::batch::matrix::dense::batch_item<const value_type>& mat,
         value_type* const __restrict__ work)
-    {}
+    {
+        work_ = work;
+        constexpr auto warp_size = config::warp_size;
+        const auto tile =
+            group::tiled_partition<warp_size>(group::this_thread_block());
+        const int tile_rank = threadIdx.x / warp_size;
+        const int num_tiles = (blockDim.x - 1) / warp_size + 1;
+        for (int irow = tile_rank; irow < mat.num_rows; irow += num_tiles) {
+            const int iz = irow * static_cast<int>(mat.stride) + irow;
+            work_[irow] = (mat.values[iz] != zero<ValueType>())
+                              ? one<ValueType>() / mat.values[iz]
+                              : one<ValueType>();
```

Comment:
you can use safe_divide here

Reply:
Safe divide sets the default to zero, but I want to set it to one

Reply:
ah, you are right